identity_stardust/src/network/network_name.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::borrow::Cow;
+
+use core::convert::TryFrom;
+use core::fmt::Display;
+use core::fmt::Formatter;
+use core::ops::Deref;
+use std::fmt::Debug;
+
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::error::Error;
+use crate::error::Result;
+
+/// Network name compliant with the [`crate::StardustDID`] method specification.
+#[derive(Clone, Hash, PartialEq, Eq, PartialOrd, Ord, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct NetworkName(pub(crate) Cow<'static, str>);
+
+impl NetworkName {
+  pub(crate) const MAX_LENGTH: usize = 6;
+  pub(crate) const DEFAULT_STR: &'static str = "main";
```

Comment:
This should be removed, it's a concern of `StardustDID` and can be accessed from `StardustDID::DEFAULT_NETWORK`.

Reply:
The `MAX_LENGTH` constant is used in `Self::validate_network_name` so I don't see how that can be moved elsewhere.  I don't quite understand what we want `NetworkName` to be. Given your comments about this so far I am wondering whether it would be better to make this generic over the DID method?

Reply:
This comment was on `DEFAULT_STR`, a previous comment was about making `MAX_LENGTH` a public associated constant.